flink-tests/src/test/java/org/apache/flink/test/checkpointing/StreamFaultToleranceTestBase.java

```diff
@@ -104,8 +106,25 @@ public void runCheckpointedProgram() {
 			postSubmit();
 		}
 		catch (Exception e) {
+			Throwable th = e;
+			int depth = 0;
+
+			for (; depth < 20; depth++) {
+				if (th instanceof SuccessException) {
+					try {
+						postSubmit();
+					} catch (Exception e1) {
+						e1.printStackTrace();
```

Comment:
Should we not forward the exception here? You introduced this block so that `postSubmit()` also runs when the `SuccessException` was thrown, right?